litert/vendors/qualcomm/core/wrappers/tests/op_wrapper_test.cc

```diff
@@ -209,5 +209,215 @@ TEST(OpWrapperTest, ChangeOpName) {
   EXPECT_STREQ(op_wrapper_1.GetOpConfig().v1.name, "namespace/name_new");
 }
 
+TensorWrapper CreateTensor(const std::vector<uint32_t>& dims, uint8_t val,
+                           Qnn_TensorType_t type = QNN_TENSOR_TYPE_APP_WRITE) {
+  std::vector<uint8_t> data(
+      std::accumulate(dims.begin(), dims.end(), 1u, std::multiplies<>()), val);
+  return TensorWrapper("", type, QNN_DATATYPE_UFIXED_POINT_8,
+                       QuantizeParamsWrapperVariant(), dims,
+                       static_cast<uint32_t>(data.size()), data.data(), true);
+}
+
+TEST(OpWrapperEqualityOperatorTest, TypeName) {
+  std::vector<uint32_t> dims = {1, 1, 3};
+  auto input = CreateTensor(dims, 1);
+  auto output = input;
```

Comment:
I think that assigning `QNN_TENSOR_TYPE_APP_WRITE` to both the input and output tensors may not be valid in QNN.

To better align with our use case, could we update the configuration so that in[0] is set to `QNN_TENSOR_TYPE_NATIVE`, in[1] to `QNN_TENSOR_TYPE_STATIC`, and out[0] to `QNN_TENSOR_TYPE_NATIVE`?